### PR TITLE
Fixes #359

### LINKED ISSUE #359: Allow travis to auto publish signed artifacts to nexus sonatype for snapshot and release builds
This is outlined in the article http://blog.eigengo.com/2014/05/25/sbt-and-oss-nexus/#more-5583 and requires the sbt-release plugin.

### PR DESCRIPTION
**Pull request by kkasravi**
Added Xiang Zhong (clockfly@gmail.com) to gpg keychain userids. Also encrypted PGPPASS=<pass phrase> in .travis.yml. If this works I'll add Tianlun and Haufeng. Note this will publish-signed and release on oss nexus. 

[Review on Reviewable](https://reviewable.io/reviews/intel-hadoop/gearpump/360)

### CODE REVIEW

**Comment by kkasravi** [comment]
Need to actually merge to master to invoke sbt clean publish-signed sonatypeRelease. Let's see what happens.

**Comment by clockfly** [comment]
@kkasravi 

To trigger a release build, it has two requisites:
1. You must create a tag on intelhadoop/gearpump,by git command line with condition
2. the version must NOT end with -SNAPSHOT

File: .travis.yml [hunk 1]
```diff
@@ -21,16 +21,17 @@ env:
...
   - secure: HLEt1PMZ8gIv1jmM7t+El8EP2I04ojPPMsjMYOfE9HxvpZ9d+xqF5M6vh43o96nq0dPLDMggZLR2WZ8DA1SSidE3+4dVGMl621kKX9wBFzenliiO8NmYpT7MisTjjSS1HLgQhCGXf77WVZVpziit1PXFR3+hSzekE8t+ZWtBvjQ=
   - secure: VyJnDR7aPtqzWmy6GX17KYI5YYGvve7kCzfng8s5/m0scIC9RhlSBv6rya7r8BFPq0OXWzZhamm44MOF+Z5Ct30vg7IpHOX6H+KwECfMoj3vV6un+OAsk+hfy5uDxGMFqMHa71W8s0URcv3p+xntjzDSGAydQsH98PMEvQ8KqG8=
+  - secure: hD/jcQkh1TTa+DxSoU7fXuJ6LT59GHPo9jFQjAIzW1WR+C8MSoFAJ26Kr3PlsuvGTt/lY827sXKATa9K/A1MP3v71AujWPs/kxusAMC66XJtV3hauT7hn2XlHKnr9FW3GTgJrkzLdPvMrvgYTk2WVKnk6SlxfUVokWPPICUatb0=
   
 after_success: 
  - codecov
    ## publish to snapshot repo
  - 'if [[ $TRAVIS_BRANCH == "master" && $TRAVIS_PULL_REQUEST == "false" && $TRAVIS_TAG == "" ]];then
-   sbt clean publish;
+   sbt clean publish-signed sonatypeRelease;
```
Comment (clockfly) [inline .travis.yml]: This is a snapshot build, it doesn't requires sonatypeRelease

Comment (kkasravi) [inline .travis.yml]: Don't we want snapshot builds available on maven? Many other repos push snapshot builds to maven.

**Pull request merged**